src/main/java/org/jenkinsci/test/acceptance/po/Jenkins.java

```diff
@@ -217,4 +230,81 @@ public JobsMixIn getJobs() {
+        URL url = url("support/generateAllBundles");
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpPost httpPost = new HttpPost(new URIBuilder()
+                    .setScheme(url.getProtocol())
+                    .setHost(url.getHost())
+                    .setPort(url.getPort())
+                    .setPath(url.getPath())
+                    .setParameter("json", supportBundleRequest.getJsonParameter())
+                    .build());
+            Crumb crumb = getCrumb(client);
+            httpPost.setHeader(crumb.getCrumbRequestField(), crumb.getCrumb());
+            try (CloseableHttpResponse response = client.execute(httpPost, buildHttpClientContext())) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                if (statusCode == 200) {
+                    HttpEntity entity = response.getEntity();
+                    if (entity != null) {
+                        try (FileOutputStream os = new FileOutputStream(supportBundleRequest.getOutputFile())) {
+                            entity.writeTo(os);
+                        }
+                    }
+                } else {
+                    throw new IOException("Got status code " + statusCode + " while getting support bundle for " + url);
+                }
+            }
+        } catch (IOException | URISyntaxException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    private Crumb getCrumb(CloseableHttpClient httpClient) throws IOException {
+        HttpGet get = new HttpGet(url("crumbIssuer/api/json").toExternalForm());
+        CloseableHttpResponse getResponse = httpClient.execute(get, buildHttpClientContext());
+        int statusCode = getResponse.getStatusLine().getStatusCode();
+        if (statusCode == 200) {
+            return new ObjectMapper().readValue(getResponse.getEntity().getContent(), Crumb.class);
+        } else {
+            throw new IOException("Got status code " + statusCode + " while getting a crumb: " + EntityUtils.toString(getResponse.getEntity()));
+        }
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class Crumb {
+        private String crumb;
+        private String crumbRequestField;
+
+        public String getCrumb() {
+            return crumb;
+        }
+
+        public void setCrumb(String crumb) {
+            this.crumb = crumb;
+        }
+
+        public String getCrumbRequestField() {
+            return crumbRequestField;
+        }
+
+        public void setCrumbRequestField(String crumbRequestField) {
+            this.crumbRequestField = crumbRequestField;
+        }
+    }
+
+    private HttpContext buildHttpClientContext() {
+        HttpClientContext context = HttpClientContext.create();
+        BasicCookieStore cookieStore = new BasicCookieStore();
+        for (org.openqa.selenium.Cookie cookie : driver.manage().getCookies()) {
```

Comment:
This adding all cookies used by `driver` up to this point. AFAICT It can contain cookies that are not from the target master here (ie. another master was accessed during the same test, which won't happen in regular Jenkins ATH but will happen in more complex ATH setups using this project as a base).

Reply:
I think there was something similar when checking for logged in users (before the recent switch to API instead of UI).

Reply:
Do you want me to change it? I see that as premature optimization.